tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -336,8 +342,12 @@ private void OnEndRequest(object sender, EventArgs eventArgs)
                         }
                         else
                         {
-                            string path = UriHelpers.GetCleanUriPath(app.Request.Url, app.Request.ApplicationPath);
-                            scope.Span.ResourceName = $"{app.Request.HttpMethod.ToUpperInvariant()} {path.ToLowerInvariant()}";
+                            var url = RequestDataHelper.GetUrl(app.Request);
+                            if (url is not null)
+                            {
+                                string path = UriHelpers.GetCleanUriPath(url, app.Request.ApplicationPath);
+                                scope.Span.ResourceName = $"{app.Request.HttpMethod.ToUpperInvariant()} {path.ToLowerInvariant()}";
+                            }
```

Comment:
I wonder if we should add some resourceName here if we cannot get the url

Reply:
Yeah, we probably should - I wonder what we should put though 😕 Unless there's an "unsafe" way we can get the URL so that we can at least include the path, I'm not sure what to do 😅 _Worst_ case we could just use `app.Request.HttpMethod.ToUpperInvariant()` - is that all we can realistically get?

Reply:
I guess that we could use reflection by calling the internal method GetUnvalidatedPath(), but that's pretty ugly and not very efficient. I have updated the code to add the method name.